### PR TITLE
Use `@types/glob` rather than the npm package

### PR DESCRIPTION
**Pull request by dolanmiu**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

**glob now has its own types in the repo. The shelljs type definition incorrectly uses glob from npm if it is installed**

```bash
node_modules/@types/shelljs/index.d.ts:1209:23 - error TS2694: Namespace '"C:/Users/Dolan/docx/node_modules/glob/dist/mjs/index"' has no exported member 'IOptions'.

1209     globOptions: glob.IOptions;
```

Notice in the above error, the `@types/shelljs` references `node_modules/glob`

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://www.npmjs.com/package/glob
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by dolanmiu** [comment]
Tests failing. A little bit stumped

**Comment by manuth** [comment]
This won't work.
`@types/` packages are added as dependencies from this repository automatically.

Take a look at the `package.json` file linked here:
https://www.npmjs.com/package/@types/shelljs?activeTab=code

`@types/glob` is already a dependency of `@types/shelljs`. The issue here is, that `glob` has a higher priority than `@types/glob`.

You might want to close this PR in favor of #64810
This PR will fix your issue once it's through

**Comment by dolanmiu** [comment]
> You might want to close this PR in favor of https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64810

@manuth No, I will leave this open as some people want to keep `@types/glob`, so I don't know what the outcome would be...

`glob` has it's own types in their own repo, so personally I think `@types/glob` should be removed, as per formal documentation.

**Comment by manuth** [comment]
I agree with you that `@types/glob` should be removed.

The point that I was trying to address is, that `@types/glob` is already listed as a dependency of `@types/shelljs`:
  - [`package.json` on unpkg](https://www.unpkg.com/browse/@types/shelljs@0.8.11/package.json)
  - [dependency list on npm](https://www.npmjs.com/package/@types/shelljs?activeTab=dependencies)

So `@types/shelljs` already uses `@types/glob`, doesn't it?

**Comment by dolanmiu** [comment]
@typescript-bot test this

**Comment by dolanmiu** [comment]
> Hey @dolanmiu,
> 
> 😒 Your PR doesn't modify any tests, so it's hard to know what's being fixed, and your changes might regress in the future. Please consider [adding tests](https://github.com/DefinitelyTyped/DefinitelyTyped#user-content-test-editing-an-existing-package) to cover the change you're making. Including tests allows this PR to be merged by yourself and the owners of this module.
> 
> _**This can potentially save days of time for you!**_

# No tests, just using the correct version of glob's types (v7) instead of `*`

**Comment by dolanmiu** [comment]
@peterblazejewicz Could you take a look?

File: types/shelljs/package.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "private": true,
+    "dependencies": {
+        "@types/glob": "^7.0.0"
```
Comment (sandersn) [inline types/shelljs/package.json:4]: unfortunately, this won't work. Even if `@types/glob` is installed, Typescript will ignore it if `glob` is installed.

**Review by sandersn — CHANGES_REQUESTED** [review]

**Comment by sandersn** [comment]
I looked at your project. I think the easiest workaround is to depend on `glob@7`/`@types/glob` in docx. Unfortunately, the way Typescript defaults to in-package types if present is a core part of how Definitely Typed works, and I'm worried that changing it would break the world.

After #64810 is in, you might also try updating `@types/shelljs` to use `glob@9` types even though `shelljs` actually uses `glob@7`. I'm not sure if that would work; it definitely wouldn't be exactly correct. But it's probably worthwhile if shelljs stops shipping updates, since it means people could use it in an up-to-date project.

**Pull request closed without merging**